docusaurus/docs/Android/02-client/06-guides/01-push-notifications.md

```diff
@@ -2,72 +2,141 @@
+  messageId = data.get("message_id"),
+  channelId = data.get("channel_id"),
+  channelType = data.get("channel_type")
+)
+ChatClient.handlePushMessage(pushMessage)
+```
 
-#### Step 1
-Go to the [Firebase Console](https://console.firebase.google.com/), and select the project your app belongs to.
+It is all you need to do client-side to integrate push notification by yourselves.  
+We also provide some [Companion Artifacts](#companion-artifacts) that help you with the integration process.
 
-#### Step 2
-Click on the gear icon next to _Project Overview_ and navigate to _Project settings_:
-![notifications step 2](../../assets/notifications_firebase_setup_step_2.jpeg)
+## PushDeviceGenerator
+Our SDK allows you to provide an implementation of `PushDeviceGenerator` and handle all the needed process to obtain and register a device into Stream.
+The `PushDeviceGenerator` is an interface that asks you to implement two method. After you implement it you need inject it into our `ChatClient`, let me show you an example:
+```kotlin
+class MyPushDeviceGenerator : PushDeviceGenerator {
+    override fun isValidForThisDevice(context: Context): Boolean {
+        // Some providers are not allowed to be used in some devices if they doesn't have the proper libraries installed on the device
+        // Here you can check if your provider could be used on this device and return true on the case it can be used on this device
+        return true
+    }
+    
+    override fun asyncGenerateDevice(onDeviceGenerated: (device: Device) -> Unit) {
+        // Here you will need to impelemnt the logic to obtain the token from your push provider and then generate the device instance
+        val token = obtainTokenFromYourPushProvider()
+        onDeviceGenerated(
+            Device(
+                token = token,
+                pushProvider = PushProvider.FIREBASE, // You can check the different Push Provider supported here: https://github.com/GetStream/stream-chat-android/blob/main/stream-chat-android-client/src/main/java/io/getstream/chat/android/client/models/Device.kt
+            )
+        )
+    }
+}
 
-#### Step 3
-Navigate to the _Cloud Messaging_ tab. Under _Project Credentials_, locate the _Server key_ and copy it:
-![notifications step 3](../../assets/notifications_firebase_setup_step_3.png)
+val notificationConfig = NotificationConfig(
+  pushDeviceGenerators = listOf(MyPushDeviceGenerator())
+)
+ChatClient.Builder("apiKey", context)
+    .notifications(ChatNotificationHandler(context, notificationsConfig))
+    .build()
+```
 
-#### Step 4
-Enable _Android & Firebase_ push notifications and upload the _Server Key_ to the Chat Dashboard:
-![notifications step 4.1](../../assets/notifications_firebase_setup_step_4_1.jpeg)
-![notifications step 4.2](../../assets/notifications_firebase_setup_step_4_2.jpeg)
+## Companion Artifacts
+The push notification integration process is not complicated, but some jobs that we can provide to you from our SDK.  
```

Comment:
`The push notification integration process is not complicated, but some jobs that we can provide to you from our SDK` This sentence is a bit hard to understand for me.